### PR TITLE
Feat: Add Audio-Based Evaluation Metrics

### PR DESCRIPTION
**Pull request by pallavkoppisetti**
### Description

This pull request introduces a new suite of metrics for evaluating audio generation models, encapsulated within the `gaico.metrics.audio` module. This provides the foundational `AudioMetric` base class and two initial, concrete implementations: **Normalized Signal-to-Noise Ratio (SNR)** and **Spectrogram Distance**.

This functionality is designed as an optional feature. All audio-related dependencies (`scipy`, `soundfile`) are imported conditionally, ensuring the core library remains lightweight for users who do not require audio evaluation. If these dependencies are not installed, the audio metrics are gracefully disabled.

### Key Implementations

*   **`AudioMetric` (Abstract Base Class):**
    *   Provides a common interface for all audio metrics.
    *   Handles multiple input types, including file paths, NumPy arrays, and lists of samples.
    *   Overrides the base `calculate` method to correctly distinguish between single-item and batch-item calculations, ensuring proper error propagation and handling.

*   **`AudioSNRNormalized`:**
    *   Calculates the Signal-to-Noise Ratio (SNR) between a generated and reference audio signal.
    *   The result is normalized to a score between 0 and 1, where 1 indicates a perfect match (no noise).

*   **`AudioSpectrogramDistance`:**
    *   Computes the similarity between two audio signals by comparing their spectrograms.
    *   Supports multiple distance functions, including `euclidean` and `correlation`.
    *   The final score is normalized to a 0-1 range, where 1 represents identical spectrograms.

*   **Comprehensive Unit Tests:**
    *   A new test file audio_metrics_test.py has been added.
    *   Includes extensive coverage for all new metrics, testing various input types, batch processing, error handling, and edge cases.

*   **Comprehensive API Documentation:** Created a new documentation page `audio.md` for the audio metrics suite. This guide provides:
    *   A detailed explanation of the calculation process for both `AudioSNRNormalized` and `AudioSpectrogramDistance`.
    *   Clear guidance on supported input formats (NumPy arrays, file paths, etc.).
    *   Practical code examples demonstrating single-item, batch, and customized usage for different applications like speech and music analysis.
   
### Future Work & To-Do

*   **Expand Metric Suite:** Implement other standard audio quality metrics
*   **Visualization Tools:** Add utility functions to plot the spectrograms used in calculations, which would be valuable for debugging and qualitative analysis.
*   **Performance Optimization:** For very large audio datasets, investigate performance enhancements, potentially through more direct use of optimized libraries or parallel processing for batch calculations.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -67,6 +67,10 @@ jsd = [
   "nltk>=3.9.1",
   "scipy>=1.15.3",
 ]
+audio = [
+    "scipy>=1.10.0",
```
[the remark this answers is not in the document]
Comment (pallavkoppisetti) [reply pyproject.toml]: I have updated the version on it to match `jsd` dependency requirement.

File: .python-version
[the remark this answers is not in the document]
Comment (g-nitin) [reply .python-version:1]: The `.python-version` file was created during the initial stages of the project ([history](https://github.com/ai4society/GenAIResultsComparator/commits/main/.python-version)), and is essential for replicability, especially within [uv](https://docs.astral.sh/uv/concepts/python-versions/#python-version-files).

As such, we would like to keep it.

**Review by g-nitin — CHANGES_REQUESTED** [review]
Everything looks good. Just one more comment (`.python-version` file) to resolve.
